src/App.jsx

```diff
@@ -14,33 +16,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABET = 'alphabet';
+const SORT_LENGTH = 'length';
+const REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState('');
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SORT_ALPHABET:
+        return good1.localeCompare(good2);
+      case SORT_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverseField !== '') {
+    visibleGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_ALPHABET)}
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_LENGTH)}
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          onClick={() => setReverseField(reverseField === '' ? REVERSE : '')}
+          className={cn('button', 'is-warning', {
+            'is-light': reverseField !== REVERSE,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(reverseField || sortField) && (
+          <button
+            type="button"
+            onClick={() => {
+              setSortField('');
+              setReverseField('');
+            }}
```

Comment:
let's create a separate function for this

Example:
```suggestion
            onClick={handleReset}
```